### ISSUE TITLE
App closing silently and very frequently since 4.14.2, 4.15.0 with ANR detection enabled.

### ISSUE THREAD

**Issue by jisung** [bugsnag/bugsnag-android#502]
### Expected behavior
App closing silently and very frequently with the below error
```
2019-06-17 22:55:38.865 5682-5716 E/WindowManager: RemoteException occurs on reporting focusChanged, w=Window{a06bc71 u0 viva.republica.toss/viva.republica.toss.send.MainActivity}
    android.os.DeadObjectException
        at android.os.BinderProxy.transactNative(Native Method)
        at android.os.BinderProxy.transact(Binder.java:1140)
        at android.view.IWindow$Stub$Proxy.windowFocusChanged(IWindow.java:500)
        at com.android.server.wm.WindowState.reportFocusChangedSerialized(WindowState.java:3888)
        at com.android.server.wm.WindowManagerService$H.handleMessage(WindowManagerService.java:5439)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:214)
        at android.os.HandlerThread.run(HandlerThread.java:65)
        at com.android.server.ServiceThread.run(ServiceThread.java:44)
```

### Observed behavior
*[Insert details on observed behaviour]*

### Steps to reproduce
Just starting the app and change tab 

### Version
4.13.0 was OK but  4.14.2, 4.15.0 have the same issues with multiple devices (Samsung Galaxy S10e, etc.)

**Comment by kattrali** [comment]
Thanks for the report, @jisung. Looking into it.

**Comment by kattrali** [comment]
Initial testing hasn't turned up anything similar, @jisung. Is there any more information you can provide about the error, including:

* How/Where bugsnag-android is being initialized in the application
* Android version
* Any other changes to dependencies other than bugsnag-android in that time span

**Comment by jisung** [comment]
**onApplicationCreate**:
```
Bugsnag.init(context, Configuration(BUGSNAG_APIKEY).apply {
            detectNdkCrashes = true
            enableExceptionHandler = true
            detectAnrs = true
        })
```
**Android version**: 9

**dependencies change**:
(from) com.bugsnag:bugsnag-android-ndk:4.13.0
(to) com.bugsnag:bugsnag-android:4.15.0

It seems to die when JNI method being called but it's not clear yet. will update you if I found something new. Thanks.

**Comment by fxdemolisher** [comment]
This happens to our project as well when using bugsnag-react-native. It started happening after the 2.21.1 upgrade of that dependency, which caused the bugsnag-android dependency to upgrade from 4.13.0 to 4.15.0.

The app shuts down silently without any stack traces or other clues in logcat. We have found a stable reproduction of the shutdown as follows:
1. Set up device to just have WIFI connectivity (mobile data off)
2. Force close the app if it was open before
3. Launch the app
4. Turn off WIFI connectivity via the notification drawer's quick settings
5. App shuts down without any clues.

If we skip this config option `config.detectAnrs = true`, the shut down does not happen. It also seems to no longer happen if the app has been open for a bit (5-10 min) after initial start up.

**Comment by fractalwrench** [comment]
Thanks for the further details @fxdemolisher - I tried following those steps on both an emulator and real device but wasn't successful in reproducing the behaviour. Would it be possible to get:

- the device model, manufacturer, and apiLevel
- logcat logs from the device
- a rough percentage of how often this behaviour can be reproduced consistently

**Comment by fxdemolisher** [comment]
@fractalwrench 

I've seen this on the Galaxy S8 (API 26/27) and S9 ( API28) test phones we have in our office as well as on a PIxel 2 (API 28). It was very reproducible right after our app was launched. The steps were to launch the app, turn off the phone's Wifi, and the app would crash silently. I did not save the logcat output snippets but will try to produce a minimal repro case and upload it here.

As for consistency. It was 100% reproducible right after app launch, but after a few minutes of being in the foreground, the rate goes down to about 1 in 5 (20%). Thats a very rough number though since we did not end up releasing to production with this version, but decided to downgrade while investigating instead.

**Comment by fxdemolisher** [comment]
Added minimal repro in: https://github.com/fxdemolisher/bugsnag-android-crash-minimal-repro

A few things that I discovered while working on the repro of the issue:
1. If detectAnrs is off in the config, the crash does not occur
2. The crash occurs if an http request is in-flight when connectivity is lost
3. Version 4.16.0 (released today) still has this issue

**Comment by fractalwrench** [comment]
Thanks very much for the detailed reproduction case @fxdemolisher - I can confirm we're able to reproduce the issue and have prioritised investigating a fix.

**Comment by fractalwrench** [comment]
A fix for this has been released in v4.16.1. Thanks to everyone for your help in reproducing this behaviour.
